### PR TITLE
Bump eventing-controller's image

### PR DESCRIPTION
**Pull request by VladislavPaskar**
update to version 1.17

### CODE REVIEW

**Review by sayanh — APPROVED** [review]

File: resources/eventing/values.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ global:
       version: PR-11920
     eventing_controller:
       name: eventing-controller
-      version: PR-12553
+      version: PR-12596
```
Comment (k15r) [inline resources/eventing/values.yaml:11]: as this image was not created in the referenced image please use either the PR-Number of the other PR (the one that creates the new image) or if that one is already merged directly refer to the final commit sha of the merging commit of the pr that created the image

File: components/eventing-controller/go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/kyma-project/kyma/components/eventing-controller
 
-go 1.16
+go 1.17
```
Comment (k15r) [inline components/eventing-controller/go.mod:3]: please run 
`go mod tidy` and update the pr. switching to go 1.17 semantics introduced require directives for indirect dependencies to be added to the go.mod file

**Review by k15r — CHANGES_REQUESTED** [review]

File: components/eventing-controller/go.mod (hunk 2 above)
Comment (VladislavPaskar) [reply components/eventing-controller/go.mod:3]: done

File: resources/eventing/values.yaml (hunk 1 above)
Comment (VladislavPaskar) [reply resources/eventing/values.yaml]: done

**Review by k15r — APPROVED** [review]

**Review by sayanh — APPROVED** [review]

**Pull request merged**